lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,206 @@
+from core_utils.article.io import to_meta, to_raw
+from core_utils.config_dto import ConfigDTO
+from core_utils.constants import (ASSETS_PATH, CRAWLER_CONFIG_PATH,
+                                  NUM_ARTICLES_UPPER_LIMIT,
+                                  TIMEOUT_LOWER_LIMIT, TIMEOUT_UPPER_LIMIT)
+
+
+class IncorrectSeedURLError(Exception):
+    """"
+    Seed URLs value is not of the list type or
+    its elements either do not match the standard URL pattern
+    or not of a string type
+    """
+    pass
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    """
+    The number of articles that must be parsed is out of given range
+    """
+    pass
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    """
+    The number of articles that must be parsed is not integer
+    """
+    pass
+
+
+class IncorrectHeadersError(Exception):
+    """
+    Headers value is not of a dictionary type
+    """
+    pass
+
+
+class IncorrectEncodingError(Exception):
+    """
+    Encoding value is not of a string type
+    """
+    pass
+
+
+class IncorrectTimeoutError(Exception):
+    """
+    Timeout value is out of range from 1 to the given value
+    """
+    pass
+
+
+class IncorrectVerifyError(Exception):
+    """
+    Should_verify_certificate value is not of a boolean type
+    """
+    pass
+
 
 class Config:
     """
     Unpacks and validates configurations
     """
 
+    seed_urls: list[str]
+    num_articles: int
+    headers: dict[str, str]
+    encoding: str
+    timeout: int
+    verify_certificate: bool
+    headless_mode: bool
+
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+
+        config_dto = self._extract_config_content()
+        self._seed_urls = config_dto.seed_urls
+        self._num_articles = config_dto.total_articles
+        self._headers = config_dto.headers
+        self._encoding = config_dto.encoding
+        self._timeout = config_dto.timeout
+        self._should_verify_certificate = config_dto.should_verify_certificate
+        self._headless_mode = config_dto.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            config_params = json.load(f)
+        return ConfigDTO(**config_params)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        config_dto = self._extract_config_content()
+
+        if not isinstance(config_dto.seed_urls, list):
+            raise IncorrectSeedURLError('seed URLs must be of passed as a list')
+
+        if not all(re.match('https?://w?w?w?.*/', url) and isinstance(url, str) for url in config_dto.seed_urls):
```

Comment:
do you think it is possible to have for example just one w?

Reply:
i don't think so, therefore it's probably better to use `w{3}?` instead of `w?w?w?`